### DISCUSSION TITLE
Used ruote Ziggy file globaly

### DISCUSSION THREAD

**Discussion by Skaywalker** [tighten/ziggy#769] [Q&A] [answered]
Hello! I'm in need of help. I don't want to use the @route expression in the blade and I want to load it globally from a generated js as it is in index.js.
The point is that I would like to put together 2 routers, one for the admin and one for the website.

Is there a way to solve it if the generated file is synchronized globally? rather, it shouldn't be used in as many trees as I use it.

![image](https://github.com/user-attachments/assets/f751510e-499a-4e04-8d00-6fca7e27888b)
![image](https://github.com/user-attachments/assets/f845d972-adce-4c47-ad61-45447d4fb60c)
![image](https://github.com/user-attachments/assets/379aeab9-2f37-44a7-ad15-86fd8c45b4bb)
here are some pictures of how I imagined the solution.
If anyone has any ideas or insights, I thank you in advance.

**Comment by bakerkretzmar** [comment, marked as answer]
I don't completely understand your screenshots but you don't need two separate routers, just two Ziggy configs. It looks like you've figured that part out, but if not there's a `--group` option you can pass to the artisan command to create separate files with separate route groups. Then everything in your first two screenshots is unnecessary—if you're using the Vue plugin, just pass it your admin config when you install it on the admin pages:

```js
// website
import { createApp } from 'vue';
import { ZiggyVue } from 'ziggy-js';
import { Ziggy } from './ziggy-website.js'; // import website routes
import App from './App.vue';

createApp(App).use(ZiggyVue, Ziggy);
```

```js
// admin
import { createApp } from 'vue';
import { ZiggyVue } from 'ziggy-js';
import { Ziggy } from './ziggy-admin.js'; // import admin routes
import App from './App.vue';

createApp(App).use(ZiggyVue, Ziggy);
```

**Reply by Skaywalker** [reply]
Thank you, only the answer was able to be solved. Of course, I checked a little bit again, I used the original route from the ziggy and not my own, so it didn't take over the config. Thanks for the confirmation.
